gem/node.py

```diff
@@ -180,6 +180,27 @@ def __call__(self, node):
             return result
 
 
+class Memoizer_cf(object):
```

Comment:
I think problem with this is that `IndexSum(Product(A, Product(B, C))`, only the outer product will be picked up?